src/parse_policies.c

```diff
@@ -82,31 +82,30 @@ int count_char_in_str(const char * string, int len, char lookup) {
 
 // parse compaction policies in the following format: "max:1m;min:10s;avg:2h;avg:3d"
 // the format is AGGREGATION_FUNCTION:\d[s|m|h|d];
-int ParseCompactionPolicy(const char * policy_string, SimpleCompactionRule **parsed_rules_out,
-            size_t *rules_count) {
+int ParseCompactionPolicy(const char *policy_string, SimpleCompactionRule **parsed_rules,
+            uint64_t *rules_count) {
     char *token;
     char *token_iter_ptr;
-    size_t len = strlen(policy_string);
+    uint64_t len = strlen(policy_string);
     char *rest = malloc(len + 1);
-    memcpy(rest, policy_string, len+1);
+    memcpy(rest, policy_string, len + 1);
     *rules_count = 0;
 
     // the ';' is a separator so we need to add +1 for the policy count
-    int policies_count = count_char_in_str(policy_string, len, ';') + 1;
```

Comment:
First we allocae and assign:
  *parsed_rules = malloc(...)
Then we go:
  (*parsed_rules)++
Then we "return" *parsed_rules.
Looks like an invitation to a leak.